sqlstore/src/main/java/uk/ac/bbsrc/tgac/miso/persistence/impl/HibernateSampleDao.java

```diff
@@ -569,7 +569,21 @@ public void setSessionFactory(SessionFactory sessionFactory) {
     criteria.add(Restrictions.or(Restrictions.ilike("externalName", str), Restrictions.ilike("alias", str)));
     @SuppressWarnings("unchecked")
     Collection<Identity> records = criteria.list();
-    // return without fetching all associated SQL items
+    // return without running fetchSqlStore() to get non-Hibernate associated entities
+    return records;
+  }
+
+  @Override
+  public Collection<Identity> getIdentitiesByExternalNameAndProject(String externalName, Long projectId) throws IOException {
+    if (isStringEmptyOrNull(externalName)) return Collections.emptySet();
+    if (projectId == null) throw new IllegalArgumentException("Must provide a projectId in search");
+    String str = DbUtils.convertStringToSearchQuery(externalName);
+    Criteria criteria = currentSession().createCriteria(IdentityImpl.class);
+    criteria.add(Restrictions.eq("project.id", projectId));
+    criteria.add(Restrictions.or(Restrictions.ilike("externalName", str), Restrictions.ilike("alias", str)));
+    @SuppressWarnings("unchecked")
+    Collection<Identity> records = criteria.list();
+    // return without running fetchSqlStore() to get non-Hibernate associated entities
```

Comment:
Because of this, an extra lookup has to be done in DefaultSampleService#findOrCreateParent. With plans to hibernatize more things, the behaviour of this method will change unexpectedly too. This should usually return 0-1 items. Doing the fetchSqlStore would avoid the extra lookup, which currently must be done whenever 1 item is found, and would provide better consistency